src/VecSim/vec_sim_tiered_index.h

```diff
@@ -33,19 +38,26 @@ class VecSimTieredIndex : public VecSimIndexInterface {
     void *jobQueueCtx; // External context to be sent to the submit callback.
     SubmitCB SubmitJobsToQueue;
 
-    void *memoryCtx;
-    UpdateMemoryCB UpdateIndexMemory;
-
     mutable std::shared_mutex flatIndexGuard;
     mutable std::shared_mutex mainIndexGuard;
 
     size_t flatBufferLimit;
 
     void submitSingleJob(AsyncJob *job) {
-        auto **jobs = array_new<AsyncJob *>(1);
-        jobs = array_append(jobs, job);
-        this->SubmitJobsToQueue(this->jobQueue, (AsyncJob **)jobs, 1, this->jobQueueCtx);
-        array_free(jobs);
+        auto destructor = AsyncJobDestructor;
+        this->SubmitJobsToQueue(this->jobQueue, this->jobQueueCtx, &job, &job->Execute, &destructor,
+                                1);
+    }
+
+    void submitJobs(vecsim_stl::vector<AsyncJob *> &jobs) {
+        vecsim_stl::vector<JobCallback> callbacks(jobs.size(), this->allocator);
+        vecsim_stl::vector<JobCallback> destructors(jobs.size(), AsyncJobDestructor,
```

Comment:
A destructor per job?

Reply:
So we can have a generic API. Each job has its own execution and it’s own destructor